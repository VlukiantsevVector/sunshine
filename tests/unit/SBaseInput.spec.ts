import { shallowMount } from '@vue/test-utils';
import SbaseInput from '@/components/SBaseInput.vue';

describe('SBaseInput', () => {
  it('doesnt show label if none is given', () => {
    const wrapper = shallowMount(SbaseInput);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows label if set', () => {
    const wrapper = shallowMount(SbaseInput, {
      propsData: {
        label: 'This is a lable',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders content in slot', () => {
    const wrapper = shallowMount(SbaseInput, {
      slots: {
        default: 'Content',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });


  it('label is big when input has no content', () => {
    const wrapper = shallowMount(SbaseInput, {
      propsData: {
        isEmpty: true,
        label: 'This is a lable',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('label is small when input has content', () => {
    const wrapper = shallowMount(SbaseInput, {
      propsData: {
        isEmpty: false,
        label: 'This is a lable',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('label is small when input has focus', () => {
    const wrapper = shallowMount(SbaseInput, {
      propsData: {
        hasFocus: true,
        label: 'This is a lable',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('label is small when input has content and focus', () => {
    const wrapper = shallowMount(SbaseInput, {
      propsData: {
        hasFocus: true,
        isEmpty: false,
        label: 'This is a lable',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
