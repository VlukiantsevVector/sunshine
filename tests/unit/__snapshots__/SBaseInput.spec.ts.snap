// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SBaseInput doesnt show label if none is given 1`] = `
<div class="s-input">
  <!---->
</div>
`;

exports[`SBaseInput label is big when input has no content 1`] = `<div class="s-input"><span class="s-input__label">This is a lable</span> </div>`;

exports[`SBaseInput label is small when input has content 1`] = `<div class="s-input"><span class="s-input__label s-input__label--aside">This is a lable</span> </div>`;

exports[`SBaseInput label is small when input has content and focus 1`] = `<div class="s-input s-input--focus"><span class="s-input__label s-input__label--aside">This is a lable</span> </div>`;

exports[`SBaseInput label is small when input has focus 1`] = `<div class="s-input s-input--focus"><span class="s-input__label s-input__label--aside">This is a lable</span> </div>`;

exports[`SBaseInput renders content in slot 1`] = `
<div class="s-input">
  <!----> Content</div>
`;

exports[`SBaseInput shows label if set 1`] = `<div class="s-input"><span class="s-input__label">This is a lable</span> </div>`;
